mindsdb/libs/data_sources/postgres_ds.py

```diff
@@ -13,37 +13,14 @@ def _setup(self, query=None, host='localhost', user='postgres', password='', dat
         if query is None:
             query = f'SELECT * FROM {table}'
 
-        con = psycopg2.connect(dbname=database, user=user, password=password, host=host, port=port)
+        con = pg8000.connect(database=database, user=user, password=password,
+                             host=host, port=port)
         df = pd.read_sql(query, con=con)
+        df.columns = [x.decode('utf-8') for x in df.columns]
```

Comment:
Hmhm, since it exhibits this weird behavior for strings it *might* just be worth it to add unit testing for a date-type to see what happens with those. And an array type if Postgres supports those by default (I forgot if it does)

Reply:
Good point, added